src/rules/noUndefinedTypes.js

```diff
@@ -191,10 +191,12 @@ export default iterateJsdoc(({
   });
 
   for (const tag of jsdocTagsWithPossibleType) {
+    const possibleType = tag.type || tag.namepathOrURL;
```

Comment:
I don't know if this is considered misuse of the `typeAllowed` "tag struct". Alternatively I could change the `tagMightHaveTypePosition` to a `tagMightHaveEitherTypeOrNamePosition` in the filter on line 190 and mark the `link` as such.

Reply:
Never mind. I tried and it failed two tests that I updated to include this case too.